python/src/scipp/_comparison.py

```diff
@@ -7,10 +7,10 @@
 
 from ._scipp import core as _cpp
 from ._cpp_wrapper_util import call_func as _call_cpp_func
-from .typing import DataArrayLike, DatasetLike
+from .typing import LabeledArray, LabeledArray
 
 
-def less(x: DataArrayLike, y: DataArrayLike) -> DataArrayLike:
+def less(x: LabeledArray, y: LabeledArray) -> LabeledArray:
```

Comment:
Are we doing ourselves a favor by introducing a 4th term, `LabeledArray`? It feels like a synonym for what a `DataArray` is, but we also mean variables and datasets? What do we gain? Isn't this adding more confusion than necessary? Or is this never displayed to the user, and instead lists the union of variable, data array, and dataset?

Reply:
With the aliases set up in `conf.py`, 'LabeledArray' is shown in the docs. I could take that out, then Sphinx expands it and shows the Union. But it would still show up in the source and help messages.